docs/docs/async-dialogs.md

```diff
@@ -4,3 +4,628 @@ title: Async Dialogs
 ---
 
 [async-dialogs](unfinished-article)
+
+Typically *promises* are used in conjunction with asynchronous tasks such as a
+network request or a `setTimeout`; a lesser explored use is in dealing with user
+input. Since a program has to wait for a user to continue some actions it makes
+sense to consider it an asynchronous event.
+
+For comparison I'll start with an example of a *synchronous* user interaction
+using the `window.prompt` and then move to an *asynchronous* interaction my
+making our own DOM based prompt. To begin here is a template for a simple HTML
+page:
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Async Dislogs Example</title>
+  <script src="//cdn.jsdelivr.net/bluebird/3.0.5/bluebird.js"></script>
```

Comment:
This should probably be the latest version of bluebird. a JSFiddle link might also be nice (an embedded snippet maybe?)